### PR TITLE
Introduce abstract base class breaking change into container registry

### PR DESCRIPTION
**Pull request by Yao725**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.ContainerRegistry.Models
+{
+    /// <summary> The UnknownContainerRegistryTaskStepProperties. </summary>
+    internal partial class UnknownContainerRegistryTaskStepProperties : ContainerRegistryTaskStepProperties
```
Comment (fengzhou-msft) [inline sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs:13]: Can we rename it to `UnknownContainerRegistryTaskStep` to align with previous naming pattern where only the base class keep the `Properties` suffix?

File: sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Azure.ResourceManager.ContainerRegistry.csproj [hunk 2]
```diff
@@ -1,8 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <Version>1.1.0-beta.2</Version>
```
Comment (fengzhou-msft) [inline sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Azure.ResourceManager.ContainerRegistry.csproj:3]: Change the version to a stable one.

File: sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs (hunk 1 above)
Comment (Yao725) [reply sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs:13]: Currently we can't rename this model as it's auto generated and I'd not like to use customization to do that. Considering it's an internal model, I think we can merge it first and I documented this issue [here](https://github.com/Azure/autorest.csharp/issues/2678#issuecomment-1245045891) to facilitate future support for this renaming in generator.

File: sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Azure.ResourceManager.ContainerRegistry.csproj (hunk 2 above)
Comment (Yao725) [reply sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Azure.ResourceManager.ContainerRegistry.csproj:3]: The version will be updated in this prepare release [PR](https://github.com/Azure/azure-sdk-for-net/pull/31090)

**Review by fengzhou-msft — APPROVED** [review]

File: sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs (hunk 1 above)
Comment (fengzhou-msft) [reply sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs:13]: I see. I didn't notice it's internal. This is added by us working as a fallback when type is not set, not swaggers?

Comment (Yao725) [reply sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Generated/Models/UnknownContainerRegistryTaskStepProperties.cs:13]: Yes, correct.

**Pull request merged**